ios/Runner/AppDelegate.swift

```diff
@@ -3,11 +3,72 @@ import Flutter
 
 @main
 @objc class AppDelegate: FlutterAppDelegate {
+  private var notificationTapEventListener: NotificationTapEventListener?
+
   override func application(
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
   ) -> Bool {
     GeneratedPluginRegistrant.register(with: self)
+    guard let controller = window?.rootViewController as? FlutterViewController else {
+      fatalError("rootViewController is not type FlutterViewController")
+    }
```

Comment:
What did you look at to determine this was the way to set up the Pigeon channels?

(I'm not finding a clear answer as I look now in the Pigeon docs.)

Reply:
I had adapted this implementation by refering to [Flutter's PlatformChannel examples](https://github.com/flutter/flutter/blob/2615ab6c1932e44e6802aaba8dc715b387ff155f/examples/platform_channel_swift/ios/Runner/AppDelegate.swift#L30-L32) and [Pigeon's example](https://github.com/flutter/packages/blob/267ac7b66308ad34b6ce14c1f5399ab0691f9ede/packages/pigeon/example/app/ios/Runner/AppDelegate.swift#L104).

Reply:
Excellent, thanks. This logic looks good, then.

For the style of it, let's use the terser `as!` form, rather than throwing with a custom error message. It should be an invariant of our app that there'll be a FlutterViewController here, and I think that's a boring enough fact that we don't need to call extra attention to it.